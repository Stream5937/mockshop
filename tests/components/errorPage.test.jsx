import { describe, it, expect } from "vitest";
import { screen } from "@testing-library/react";
import { renderWithRouter } from "../utils/renderWithRouter.jsx";
import ErrorPage from "../../src/components/ErrorPage.jsx";

//to test the test process!
describe("something truthy and falsy", () => {
  it("true to be true", () => {
    expect(true).toBe(true);
  });

  it("false to be false", () => {
    expect(false).toBe(false);
  });
});

describe("Error page component", () => {
  it("renders", () => {
    const { asFragment } = renderWithRouter(<ErrorPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders text", () => {
    renderWithRouter(<ErrorPage />);
    expect(screen.getByRole("heading").textContent).toMatch(
      /Oh no, this route doesn't exist!/i
    );
  });
});
