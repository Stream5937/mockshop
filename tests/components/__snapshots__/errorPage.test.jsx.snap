// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error page component > renders 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Oh no, this route doesn't exist!
    </h1>
    <a
      data-discover="true"
      href="/"
    >
      You can go back to the home page by clicking here, though!
    </a>
  </div>
</DocumentFragment>
`;
